.github/workflows/go-tests.yml

```diff
@@ -0,0 +1,66 @@
+name: go-tests
+
+on: [push]
+
+env:
+  TEST_RESULTS: /tmp/test-results
+
+jobs:
+
+  go-tests:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        go-version: [1.19]
+    
+    steps:
+      - name: Setup go
+        uses: actions/setup-go@v3
+        with:
+          go-version: ${{ matrix.go-version }}
+      
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Create test directory
+        run: |
+          mkdir -p ${{env.TEST_RESULTS}}
+
+      - name: Download go modules
+        run: go mod download
+      
+      - name: Cache / restore go modules
+        uses: actions/cache@v3
+        with:
+          path: |
+            ~/.cache/go-build
```

Comment:
Previously we only cached `~/go/pkg/mod` just wondering why we're also caching the go build cache as well in this change? I can see the logic in wanting to, but I'm wondering if it's useful to do this unless we also update the key to take into account the go version? Also, since Go doesn't automatically trim the build cache, it's likely to keep growing larger over time due to the `restore-keys` being set to always restore something even without an exact cache hit based on `go.sum`. For simplicity I would favour not caching this directory, unless it has a significant impact on build time, in which case we should cache it more conservatively, separate from the mod cache, based only on exact cache hits based on `go.sum` and the go version. Hope that makes sense, happy to talk through any of this if not! :)

Reply:
Ah this makes total sense -- I was actually following the format that was suggested in the cache action repo, [here](https://github.com/actions/cache/blob/main/examples.md#go---modules) +1 to removing for now as it sounds like it would just bog things down, but can definitely revisit this in the future if its needed.

Reply:
Ah, interesting, thanks for the link to the docs... In the short term for a big project it might make things faster, but I doubt the impact will be very significant for this repo even in the short term when the cache size is small.